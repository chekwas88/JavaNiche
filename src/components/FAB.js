import * as React from 'react';
import { StyleSheet, Share } from 'react-native';
import { FAB } from 'react-native-paper';

const styles = StyleSheet.create({
  fab: {
    position: 'absolute',
    margin: 16,
    right: 0,
    bottom: 0
  }
});

class SocialShare extends React.Component {
  onShare = async () => {
    const { userName, profileUrl } = this.props;
    try {
      const result = await Share.share({
        message: `Check out this awesome developer @${userName}, ${profileUrl}`
      });
    } catch (error) {
      alert(error.message);
    }
  };

  render() {
    return <FAB style={styles.fab} small icon="add" onPress={this.onShare} />;
  }
}
export default SocialShare;
