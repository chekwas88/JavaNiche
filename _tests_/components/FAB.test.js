import React from 'react';
import { shallow } from 'enzyme';
import FAB from '../../src/components/FAB';

describe('Test for FAB component', () => {
  const wrapper = shallow(<FAB />);
  test('should test for login ', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
describe('Test for share functionality', () => {
  it('should call onPress', () => {
    const wrapper = shallow(<FAB />);
    const shareButton = wrapper.find('withTheme(FAB)');
    const shareInstance = wrapper.instance();
    shareInstance.onShare();
    const mockOnPress = (shareInstance.onShare = jest.fn());
    shareButton.simulate('onPress', mockOnPress());
    expect(mockOnPress).toHaveBeenCalled();
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
