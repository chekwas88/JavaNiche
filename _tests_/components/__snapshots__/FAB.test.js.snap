// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for FAB component should test for login  1`] = `ShallowWrapper {}`;
